tests/pytests/test_empty.py

```diff
@@ -0,0 +1,655 @@
+from common import *
+import json
+
+EMPTY_RESULT = [0]
+
+def testEmptyValidations():
+    """Validates the edge-cases of the `ISEMPTY` field option"""
+
+    env = Env(moduleArgs="DEFAULT_DIALECT 2")
+    conn = getConnectionByEnv(env)
+
+    # Create an index with a TAG and a TEXT field, both of which don't index
+    # empty values
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'tag', 'TAG', 'text', 'TEXT').ok()
+
+    # Test that we get in error in case of a user tries to use "isempty(@field)"
+    # when `field` does not index empty values.
+    env.expect('FT.SEARCH', 'idx', 'isempty(@tag)').error().contains('does not index empty values')
+    env.expect('FT.SEARCH', 'idx', 'isempty(@text)').error().contains('does not index empty values')
+    # Empty search on a non-existing field
+    env.expect('FT.SEARCH', 'idx', 'isempty(@non_existing)').error().contains('Field not found')
+
+
+def EmptyJSONTest(env, idx):
```

Comment:
Consider renaming to something like EmptyJSONTagTest?
Or make it a an inner function as well? (like `testEmptyTextHash`)

What about empty text with JSON?

(there's `testEmptyTextHash` and (suggested) `testEmptyTagHash`)

Reply:
There is a test for empty text with JSON as well, this is the reason I took this funciton outside the tag case.